mewarpx/mewarpx/emission.py

```diff
@@ -1277,6 +1414,148 @@ def get_normals(self, x, y, z):
         return normals
 
 
+class ZCylinderEmitter(Emitter):
+
+    """This injects over the side faces of a cylinder oriented along z."""
+    geoms = ['RZ', 'XYZ']
+
+    def __init__(self, conductor, T=None, zmin=None, zmax=None, rsign=1,
+                 transverse_fac=1.0, **kwargs):
+        """Initialize a 3D cylindrical emitter oriented along the z-axis.
+
+        Arguments:
+            conductor (:class:`mewarpx.assemblies.Assembly`): Conductor object,
+                used to obtain work function, coordinates and possibly
+                temperature.
+            T (float): Temperature in Kelvin for the emitter; determines
+                velocities. If not specified the temperature of the conductor
+                will be used.
+            zmin (float): Lower z-coordinate for emitting surface. Default
+                conductor.zmin if it exists otherwise mwxrun.zmin.
+            zmax (float): Upper z-coordinate for emitting surface. Default
+                conductor.zmax if it exists otherwise mwxrun.zmax.
+            rsign (float): 1 for emitting outward of the cylinder (will use
+                r_outer attribute of the conductor for the emitting surface),
+                -1 for emitting inward towards r = 0 (will use r_inner attribute
+                of the conductor for the emitting surface). Default 1.
+            transverse_fac (float): Scale the transverse energy distribution by
+                this factor. Default 1. See
+                :func:`mewarpx.utils_store.util.get_velocities` for details.
+            kwargs (dict): Any other keyword arguments supported by the parent
+                Emitter constructor (such as "emission_type").
+
+        Notes:
+            The center of the cylinder is always x = y = 0 at present.
+        """
+        # Default initialization
+        super(ZCylinderEmitter, self).__init__(T=T, conductor=conductor,
+                                               **kwargs)
+
+        self.zmin = zmin
+        if self.zmin is None:
+            if hasattr(conductor, 'zmin'):
+                self.zmin = conductor.zmin
+            else:
+                self.zmin = mwxrun.zmin
+        self.zmax = zmax
+        if self.zmax is None:
+            if hasattr(conductor, 'zmax'):
+                self.zmax = conductor.zmax
+            else:
+                self.zmax = mwxrun.zmax
+
+        self.rsign = int(round(rsign))
+        if self.rsign not in [-1, 1]:
+            raise ValueError("rsign must be +1 or -1 for z-cylinder emitters.")
+        if self.rsign == 1:
+            self.r = conductor.r_outer + 1e-10
+        else:
+            self.r = conductor.r_inner - 1e-10
+        self.transverse_fac = transverse_fac
+
+        # sanity check
+        if self.r == 0:
```

Comment:
```suggestion
        if self.r <= 0:
```
Otherwise the `- 1e-10` just above will cause this check to be skipped for `r_inner = 0`.